operations/helm/scripts/build.sh

```diff
@@ -73,12 +73,20 @@ for FILEPATH in $TESTS; do
   fi
 
   set -x
-  helm template "${ARGS[@]}"
+  helm template "${ARGS[@]}" &
   set +x
+done
+
+wait
```

Comment:
What happens if one of the background `helm template` jobs fails? The [docs for `wait`](https://ss64.com/bash/wait.html) say that using `wait` like this will always return a zero exit code:

> If n is not given, all currently active child processes are waited for, and the return status is zero.

Reply:
oh, nice catch. It took way longer than I'd like to admit to figure out how to exit when a child processes has exited with a non-zero code. I think I finally got it in https://github.com/grafana/mimir/pull/5354/commits/71e0bdfdf5c8cada587ca0f3ecfce3764d61f35f

I also rejigged the error messages and discarded the stdout of `helm template` to make it a bit easier to find what's failed.

Reply:
> It took way longer than I'd like to admit to figure out how to exit when a child processes has exited with a non-zero code.

Yeah, Bash requires very specific incantations to do seemingly simple things 😄 

> I also rejigged the error messages and discarded the stdout of `helm template` to make it a bit easier to find what's failed.

Does this mean that we don't see the full error message from `helm template` if it fails? And if multiple invocations fail, will they all compete to write to stderr at the same time and have their output interleaved?

I'm also wondering if we should wait for all `helm template` invocations to complete, even if one fails, and then exit if any failed after they're all finished - otherwise later invocations might continue running in the background and produce output at unexpected times (when running locally) or be lost altogether (when running on CI).

Reply:
> Does this mean that we don't see the full error message from helm template if it fails? And if multiple invocations fail, will they all compete to write to stderr at the same time and have their output interleaved?

yes, I'm afraid that will happen. I can attempt to write stderr of each subcommand into a file and the print the contents of all such files, but I think that's becoming very complex for a CI job.




> I'm also wondering if we should wait for all `helm template` invocations to complete, even if one fails, and then exit if any failed after they're all finished - otherwise later invocations might continue running in the background and produce output at unexpected times (when running locally) or be lost altogether (when running on CI).

good point. I did a change in 18c5b5a41f73218c40ce1d2af6fcba3835d699d2 and this is the current output if one helm template fails

<details><summary>Details</summary>

```
docker pull grafana/mimir-build-image:update-golangci-lint-d48f951b3
update-golangci-lint-d48f951b3: Pulling from grafana/mimir-build-image
Digest: sha256:fcd642527dec1456a338fdb10aa6a7d4af1134dee9a8cbb12502b641d1dbc178
Status: Image is up to date for grafana/mimir-build-image:update-golangci-lint-d48f951b3
docker.io/grafana/mimir-build-image:update-golangci-lint-d48f951b3
docker tag grafana/mimir-build-image:update-golangci-lint-d48f951b3 grafana/mimir-build-image:latest
touch mimir-build-image/.uptodate
>>>> Entering build container: build-helm-tests
time docker run --rm --tty -i -v ~/.ssh/:/root/.ssh:delegated,z -v mimir-go-cache:/go/cache -v mimir-go-pkg:/go/pkg -v /Users/dimitar/grafana/mimir:/go/src/github.com/grafana/mimir:delegated,z grafana/mimir-build-image GOOS=darwin GOARCH=arm64 BINARY_SUFFIX="" build-helm-tests;
make: Entering directory '/go/src/github.com/grafana/mimir'
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
0
Helm dependencies are up to date

Templating test-oss-logical-multizone-values
Checking for kubeVersionOverride inside tests' values.yaml ...

Templating test-oss-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"
Launching helm template in PID 62 'helm template test-oss-logical-multizone-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/test-oss-logical-multizone-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-logical-multizone-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating test-enterprise-configmap-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"
Launching helm template in PID 65 'helm template test-oss-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/test-oss-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating test-oss-multizone-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"
Launching helm template in PID 68 'helm template test-enterprise-configmap-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/test-enterprise-configmap-values.yaml --output-dir operations/helm/manifests-intermediate/test-enterprise-configmap-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating graphite-enabled-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"

Templating large-values
Launching helm template in PID 73 'helm template test-oss-multizone-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-oss-multizone-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-multizone-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"
Launching helm template in PID 80 'helm template graphite-enabled-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/graphite-enabled-values.yaml --output-dir operations/helm/manifests-intermediate/graphite-enabled-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating small-values
Checking for kubeVersionOverride inside tests' values.yaml ...

Templating test-oss-topology-spread-constraints-values
Checking for kubeVersionOverride inside tests' values.yaml ...

Templating metamonitoring-values
Checking for kubeVersionOverride inside tests' values.yaml ...
Warning: injecting Kubernetes version override: kubeVersionOverride=1.20
Launching helm template in PID 88 'helm template large-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/large-values.yaml --output-dir operations/helm/manifests-intermediate/large-values-generated --namespace citestns --set-string regoTestGenerateValues=true --set-string kubeVersionOverride=1.20'
kubeVersionOverride: "1.20"
kubeVersionOverride: "1.20"
Warning: injecting Kubernetes version override: kubeVersionOverride=1.20
Launching helm template in PID 96 'helm template small-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/small-values.yaml --output-dir operations/helm/manifests-intermediate/small-values-generated --namespace citestns --set-string regoTestGenerateValues=true --set-string kubeVersionOverride=1.20'
Launching helm template in PID 107 'helm template metamonitoring-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/metamonitoring-values.yaml --output-dir operations/helm/manifests-intermediate/metamonitoring-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
Launching helm template in PID 100 'helm template test-oss-topology-spread-constraints-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-oss-topology-spread-constraints-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-topology-spread-constraints-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating test-vault-agent-values
Checking for kubeVersionOverride inside tests' values.yaml ...

Templating test-enterprise-legacy-label-values
Checking for kubeVersionOverride inside tests' values.yaml ...

Templating test-oss-k8s-1.25-values
Checking for kubeVersionOverride inside tests' values.yaml ...

Templating scheduler-name-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"
kubeVersionOverride: "1.20"

Templating gateway-nginx-values
Checking for kubeVersionOverride inside tests' values.yaml ...
Launching helm template in PID 137 'helm template test-enterprise-legacy-label-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-enterprise-legacy-label-values.yaml --output-dir operations/helm/manifests-intermediate/test-enterprise-legacy-label-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
Launching helm template in PID 111 'helm template test-vault-agent-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-vault-agent-values.yaml --output-dir operations/helm/manifests-intermediate/test-vault-agent-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating openshift-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.25"

Templating gateway-enterprise-values
Checking for kubeVersionOverride inside tests' values.yaml ...
Launching helm template in PID 143 'helm template test-oss-k8s-1.25-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-oss-k8s-1.25-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-k8s-1.25-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
kubeVersionOverride: "1.20"
Launching helm template in PID 153 'helm template gateway-nginx-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/gateway-nginx-values.yaml --output-dir operations/helm/manifests-intermediate/gateway-nginx-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
kubeVersionOverride: "1.20"
Launching helm template in PID 155 'helm template openshift-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/openshift-values.yaml --output-dir operations/helm/manifests-intermediate/openshift-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
kubeVersionOverride: "1.20"
Launching helm template in PID 149 'helm template scheduler-name-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/scheduler-name-values.yaml --output-dir operations/helm/manifests-intermediate/scheduler-name-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used

Templating test-enterprise-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"
Launching helm template in PID 191 'helm template test-enterprise-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-enterprise-values.yaml --output-dir operations/helm/manifests-intermediate/test-enterprise-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating test-enterprise-k8s-1.25-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.25"
Launching helm template in PID 202 'helm template test-enterprise-k8s-1.25-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-enterprise-k8s-1.25-values.yaml --output-dir operations/helm/manifests-intermediate/test-enterprise-k8s-1.25-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
kubeVersionOverride: "wpeoighjrperiogjhoepgir"
Launching helm template in PID 162 'helm template gateway-enterprise-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/gateway-enterprise-values.yaml --output-dir operations/helm/manifests-intermediate/gateway-enterprise-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.admin_api.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.results-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.alertmanager.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.querier.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.ingester.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.admin-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.chunks-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.gateway.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.metadata-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.compactor.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.index-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.query_frontend.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.query_scheduler.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.store_gateway.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.overrides_exporter.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.results-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.index-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.alertmanager.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.gateway.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.metadata-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.compactor.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.admin_api.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.query_scheduler.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.ingester.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.query_frontend.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.overrides_exporter.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.querier.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.admin-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.store_gateway.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.chunks-cache.topologySpreadConstraints (map[])
Error: template: mimir-distributed/templates/store-gateway/store-gateway-pdb.yaml:1:4: executing "mimir-distributed/templates/store-gateway/store-gateway-pdb.yaml" at <include "mimir.lib.podDisruptionBudget" (dict "ctx" $ "component" "store-gateway" "memberlist" true)>: error calling include: template: mimir-distributed/templates/lib/pdb.tpl:10:15: executing "mimir.lib.podDisruptionBudget" at <include "mimir.podDisruptionBudget.apiVersion" $.ctx>: error calling include: template: mimir-distributed/templates/_helpers.tpl:381:9: executing "mimir.podDisruptionBudget.apiVersion" at <semverCompare ">= 1.21-0" (include "mimir.kubeVersion" .)>: error calling semverCompare: Invalid Semantic Version

Use --debug flag to render out invalid YAML
helm template PID 162 exited with non-zero exit code 1. Aborting.
make: *** [Makefile:449: build-helm-tests] Error 1
make: Leaving directory '/go/src/github.com/grafana/mimir'

real	0m7.372s
user	0m0.015s
sys	0m0.028s
make: *** [build-helm-tests] Error 2
```

</details>

Reply:
and this is what it looks like with two failues

<details><summary>Details</summary>

```
docker pull grafana/mimir-build-image:update-golangci-lint-d48f951b3
update-golangci-lint-d48f951b3: Pulling from grafana/mimir-build-image
Digest: sha256:fcd642527dec1456a338fdb10aa6a7d4af1134dee9a8cbb12502b641d1dbc178
Status: Image is up to date for grafana/mimir-build-image:update-golangci-lint-d48f951b3
docker.io/grafana/mimir-build-image:update-golangci-lint-d48f951b3
docker tag grafana/mimir-build-image:update-golangci-lint-d48f951b3 grafana/mimir-build-image:latest
touch mimir-build-image/.uptodate
>>>> Entering build container: build-helm-tests
time docker run --rm --tty -i -v ~/.ssh/:/root/.ssh:delegated,z -v mimir-go-cache:/go/cache -v mimir-go-pkg:/go/pkg -v /Users/dimitar/grafana/mimir:/go/src/github.com/grafana/mimir:delegated,z grafana/mimir-build-image GOOS=darwin GOARCH=arm64 BINARY_SUFFIX="" build-helm-tests;
make: Entering directory '/go/src/github.com/grafana/mimir'
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
0
Helm dependencies are up to date

Templating test-oss-logical-multizone-values
Checking for kubeVersionOverride inside tests' values.yaml ...

Templating test-oss-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"
Launching helm template in PID 62 'helm template test-oss-logical-multizone-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/test-oss-logical-multizone-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-logical-multizone-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating test-enterprise-configmap-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"
Launching helm template in PID 65 'helm template test-oss-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/test-oss-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating test-oss-multizone-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"
Launching helm template in PID 68 'helm template test-enterprise-configmap-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/test-enterprise-configmap-values.yaml --output-dir operations/helm/manifests-intermediate/test-enterprise-configmap-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
kubeVersionOverride: "1.20"

Templating graphite-enabled-values
Checking for kubeVersionOverride inside tests' values.yaml ...
Launching helm template in PID 72 'helm template test-oss-multizone-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-oss-multizone-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-multizone-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating large-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"

Templating small-values
Checking for kubeVersionOverride inside tests' values.yaml ...
Launching helm template in PID 80 'helm template graphite-enabled-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/graphite-enabled-values.yaml --output-dir operations/helm/manifests-intermediate/graphite-enabled-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating metamonitoring-values
Checking for kubeVersionOverride inside tests' values.yaml ...
Warning: injecting Kubernetes version override: kubeVersionOverride=1.20
Launching helm template in PID 87 'helm template large-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/large-values.yaml --output-dir operations/helm/manifests-intermediate/large-values-generated --namespace citestns --set-string regoTestGenerateValues=true --set-string kubeVersionOverride=1.20'

Templating test-vault-agent-values
Checking for kubeVersionOverride inside tests' values.yaml ...

Templating test-oss-topology-spread-constraints-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"

Templating test-enterprise-legacy-label-values
Checking for kubeVersionOverride inside tests' values.yaml ...

Templating test-oss-k8s-1.25-values
Checking for kubeVersionOverride inside tests' values.yaml ...
Warning: injecting Kubernetes version override: kubeVersionOverride=1.20
Launching helm template in PID 94 'helm template small-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/small-values.yaml --output-dir operations/helm/manifests-intermediate/small-values-generated --namespace citestns --set-string regoTestGenerateValues=true --set-string kubeVersionOverride=1.20'
kubeVersionOverride: "1.20"
Launching helm template in PID 106 'helm template metamonitoring-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/metamonitoring-values.yaml --output-dir operations/helm/manifests-intermediate/metamonitoring-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
kubeVersionOverride: "1.20"
Launching helm template in PID 112 'helm template test-vault-agent-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-vault-agent-values.yaml --output-dir operations/helm/manifests-intermediate/test-vault-agent-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating scheduler-name-values
Checking for kubeVersionOverride inside tests' values.yaml ...
Launching helm template in PID 102 'helm template test-oss-topology-spread-constraints-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-oss-topology-spread-constraints-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-topology-spread-constraints-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
kubeVersionOverride: "1.25"

Templating gateway-nginx-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.20"
kubeVersionOverride: "1.20"
Launching helm template in PID 124 'helm template test-oss-k8s-1.25-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-oss-k8s-1.25-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-k8s-1.25-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
Launching helm template in PID 129 'helm template scheduler-name-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/scheduler-name-values.yaml --output-dir operations/helm/manifests-intermediate/scheduler-name-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating openshift-values
Checking for kubeVersionOverride inside tests' values.yaml ...
Launching helm template in PID 121 'helm template test-enterprise-legacy-label-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-enterprise-legacy-label-values.yaml --output-dir operations/helm/manifests-intermediate/test-enterprise-legacy-label-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
kubeVersionOverride: "1.20"
kubeVersionOverride: "wpeoighjrperiogjhoepgir"

Templating gateway-enterprise-values
Checking for kubeVersionOverride inside tests' values.yaml ...
Launching helm template in PID 149 'helm template openshift-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/openshift-values.yaml --output-dir operations/helm/manifests-intermediate/openshift-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
Launching helm template in PID 140 'helm template gateway-nginx-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/gateway-nginx-values.yaml --output-dir operations/helm/manifests-intermediate/gateway-nginx-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating test-enterprise-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "wpeoighjrperiogjhoepgir"
kubeVersionOverride: "1.20"
Launching helm template in PID 156 'helm template gateway-enterprise-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/gateway-enterprise-values.yaml --output-dir operations/helm/manifests-intermediate/gateway-enterprise-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
Launching helm template in PID 184 'helm template test-enterprise-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-enterprise-values.yaml --output-dir operations/helm/manifests-intermediate/test-enterprise-values-generated --namespace citestns --set-string regoTestGenerateValues=true'

Templating test-enterprise-k8s-1.25-values
Checking for kubeVersionOverride inside tests' values.yaml ...
kubeVersionOverride: "1.25"
Launching helm template in PID 194 'helm template test-enterprise-k8s-1.25-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-enterprise-k8s-1.25-values.yaml --output-dir operations/helm/manifests-intermediate/test-enterprise-k8s-1.25-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.chunks-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.query_scheduler.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.admin-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.gateway.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.alertmanager.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.compactor.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.index-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.metadata-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.query_frontend.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.admin_api.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.ingester.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.overrides_exporter.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.results-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.querier.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.store_gateway.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.metadata-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.overrides_exporter.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.store_gateway.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.chunks-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.compactor.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.query_scheduler.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.admin_api.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.ingester.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.admin-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.alertmanager.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.query_frontend.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.results-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.querier.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.index-cache.topologySpreadConstraints (map[])
coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.gateway.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
Error: template: mimir-distributed/templates/store-gateway/store-gateway-pdb.yaml:1:4: executing "mimir-distributed/templates/store-gateway/store-gateway-pdb.yaml" at <include "mimir.lib.podDisruptionBudget" (dict "ctx" $ "component" "store-gateway" "memberlist" true)>: error calling include: template: mimir-distributed/templates/lib/pdb.tpl:10:15: executing "mimir.lib.podDisruptionBudget" at <include "mimir.podDisruptionBudget.apiVersion" $.ctx>: error calling include: template: mimir-distributed/templates/_helpers.tpl:381:9: executing "mimir.podDisruptionBudget.apiVersion" at <semverCompare ">= 1.21-0" (include "mimir.kubeVersion" .)>: error calling semverCompare: Invalid Semantic Version

Use --debug flag to render out invalid YAML
Error: template: mimir-distributed/templates/store-gateway/store-gateway-pdb.yaml:1:4: executing "mimir-distributed/templates/store-gateway/store-gateway-pdb.yaml" at <include "mimir.lib.podDisruptionBudget" (dict "ctx" $ "component" "store-gateway" "memberlist" true)>: error calling include: template: mimir-distributed/templates/lib/pdb.tpl:10:15: executing "mimir.lib.podDisruptionBudget" at <include "mimir.podDisruptionBudget.apiVersion" $.ctx>: error calling include: template: mimir-distributed/templates/_helpers.tpl:381:9: executing "mimir.podDisruptionBudget.apiVersion" at <semverCompare ">= 1.21-0" (include "mimir.kubeVersion" .)>: error calling semverCompare: Invalid Semantic Version

Use --debug flag to render out invalid YAML
helm template PID 140 exited with non-zero exit code 1. Aborting.
make: *** [Makefile:449: build-helm-tests] Error 1
make: Leaving directory '/go/src/github.com/grafana/mimir'

real	0m7.047s
user	0m0.016s
sys	0m0.029s
make: *** [build-helm-tests] Error 2
```

</details>

Reply:
Looking good

Could we do some Bash magic to prefix the output from each invocation? Something like this perhaps?

`generate_manifests $TEST_NAME $INTERMEDIATE_OUTPUT_DIR $OUTPUT_DIR > >(sed "s/^/$TEST_NAME: /") 2> >(sed "s/^/$TEST_NAME (err): /" >&2) &`

(or possibly even just redirect stderr to stdout and then apply the prefix to stdout)

Reply:
oh that's really neat! I didn't realize it will be that easy. Here's the output after the latest change

<details><summary>Details</summary>
<p>

```
make build-helm-tests  | grep -v 'Permission denied'
docker pull grafana/mimir-build-image:update-golangci-lint-d48f951b3
update-golangci-lint-d48f951b3: Pulling from grafana/mimir-build-image
Digest: sha256:fcd642527dec1456a338fdb10aa6a7d4af1134dee9a8cbb12502b641d1dbc178
Status: Image is up to date for grafana/mimir-build-image:update-golangci-lint-d48f951b3
docker.io/grafana/mimir-build-image:update-golangci-lint-d48f951b3
docker tag grafana/mimir-build-image:update-golangci-lint-d48f951b3 grafana/mimir-build-image:latest
touch mimir-build-image/.uptodate
>>>> Entering build container: build-helm-tests
time docker run --rm --tty -i -v ~/.ssh/:/root/.ssh:delegated,z -v mimir-go-cache:/go/cache -v mimir-go-pkg:/go/pkg -v /Users/dimitar/grafana/mimir:/go/src/github.com/grafana/mimir:delegated,z grafana/mimir-build-image GOOS=darwin GOARCH=arm64 BINARY_SUFFIX="" build-helm-tests;
make: Entering directory '/go/src/github.com/grafana/mimir'
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
0
Helm dependencies are up to date
test-oss-logical-multizone-values (stdout):
test-oss-logical-multizone-values (stdout): Templating test-oss-logical-multizone-values
test-oss-logical-multizone-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-oss-logical-multizone-values (stdout): kubeVersionOverride: "1.20"
test-oss-values (stdout):
test-oss-values (stdout): Templating test-oss-values
test-oss-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-oss-logical-multizone-values (stdout): Rendering helm test test-oss-logical-multizone-values in PID 63: 'helm template test-oss-logical-multizone-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/test-oss-logical-multizone-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-logical-multizone-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
test-enterprise-configmap-values (stdout):
test-enterprise-configmap-values (stdout): Templating test-enterprise-configmap-values
test-enterprise-configmap-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-oss-values (stdout): kubeVersionOverride: "1.20"
test-oss-values (stdout): Rendering helm test test-oss-values in PID 70: 'helm template test-oss-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/test-oss-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
test-enterprise-configmap-values (stdout): kubeVersionOverride: "1.20"
test-oss-multizone-values (stdout):
test-oss-multizone-values (stdout): Templating test-oss-multizone-values
test-oss-multizone-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-enterprise-configmap-values (stdout): Rendering helm test test-enterprise-configmap-values in PID 77: 'helm template test-enterprise-configmap-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/test-enterprise-configmap-values.yaml --output-dir operations/helm/manifests-intermediate/test-enterprise-configmap-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
test-oss-multizone-values (stdout): kubeVersionOverride: "1.20"
test-oss-multizone-values (stdout): Rendering helm test test-oss-multizone-values in PID 85: 'helm template test-oss-multizone-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-oss-multizone-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-multizone-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
graphite-enabled-values (stdout):
graphite-enabled-values (stdout): Templating graphite-enabled-values
graphite-enabled-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
graphite-enabled-values (stdout): kubeVersionOverride: "1.20"
graphite-enabled-values (stdout): Rendering helm test graphite-enabled-values in PID 97: 'helm template graphite-enabled-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/graphite-enabled-values.yaml --output-dir operations/helm/manifests-intermediate/graphite-enabled-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
large-values (stdout):
large-values (stdout): Templating large-values
large-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
small-values (stdout):
small-values (stdout): Templating small-values
small-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
large-values (stdout): Warning: injecting Kubernetes version override: kubeVersionOverride=1.20
large-values (stdout): Rendering helm test large-values in PID 125: 'helm template large-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/large-values.yaml --output-dir operations/helm/manifests-intermediate/large-values-generated --namespace citestns --set-string regoTestGenerateValues=true --set-string kubeVersionOverride=1.20'
test-oss-topology-spread-constraints-values (stdout):
small-values (stdout): Warning: injecting Kubernetes version override: kubeVersionOverride=1.20
small-values (stdout): Rendering helm test small-values in PID 133: 'helm template small-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/small-values.yaml --output-dir operations/helm/manifests-intermediate/small-values-generated --namespace citestns --set-string regoTestGenerateValues=true --set-string kubeVersionOverride=1.20'
metamonitoring-values (stdout):
metamonitoring-values (stdout): Templating metamonitoring-values
metamonitoring-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-oss-topology-spread-constraints-values (stdout): Templating test-oss-topology-spread-constraints-values
test-oss-topology-spread-constraints-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-vault-agent-values (stdout):
test-vault-agent-values (stdout): Templating test-vault-agent-values
test-vault-agent-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-oss-k8s-1.25-values (stdout):
test-oss-k8s-1.25-values (stdout): Templating test-oss-k8s-1.25-values
test-oss-k8s-1.25-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-oss-k8s-1.25-values (stdout): kubeVersionOverride: "1.25"
metamonitoring-values (stdout): kubeVersionOverride: "1.20"
metamonitoring-values (stdout): Rendering helm test metamonitoring-values in PID 159: 'helm template metamonitoring-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/metamonitoring-values.yaml --output-dir operations/helm/manifests-intermediate/metamonitoring-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
test-oss-k8s-1.25-values (stdout): Rendering helm test test-oss-k8s-1.25-values in PID 175: 'helm template test-oss-k8s-1.25-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-oss-k8s-1.25-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-k8s-1.25-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
test-vault-agent-values (stdout): kubeVersionOverride: "1.20"
test-vault-agent-values (stdout): Rendering helm test test-vault-agent-values in PID 165: 'helm template test-vault-agent-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-vault-agent-values.yaml --output-dir operations/helm/manifests-intermediate/test-vault-agent-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
gateway-nginx-values (stdout):
gateway-nginx-values (stdout): Templating gateway-nginx-values
gateway-nginx-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-oss-topology-spread-constraints-values (stdout): kubeVersionOverride: "1.20"
openshift-values (stdout):
openshift-values (stdout): Templating openshift-values
openshift-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-oss-topology-spread-constraints-values (stdout): Rendering helm test test-oss-topology-spread-constraints-values in PID 136: 'helm template test-oss-topology-spread-constraints-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-oss-topology-spread-constraints-values.yaml --output-dir operations/helm/manifests-intermediate/test-oss-topology-spread-constraints-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
gateway-nginx-values (stdout): kubeVersionOverride: "wpeoighjrperiogjhoepgir"
gateway-nginx-values (stdout): Rendering helm test gateway-nginx-values in PID 198: 'helm template gateway-nginx-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/gateway-nginx-values.yaml --output-dir operations/helm/manifests-intermediate/gateway-nginx-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
openshift-values (stdout): kubeVersionOverride: "1.20"
openshift-values (stdout): Rendering helm test openshift-values in PID 206: 'helm template openshift-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/openshift-values.yaml --output-dir operations/helm/manifests-intermediate/openshift-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
test-enterprise-legacy-label-values (stdout):
test-enterprise-legacy-label-values (stdout): Templating test-enterprise-legacy-label-values
test-enterprise-legacy-label-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-enterprise-legacy-label-values (stdout): kubeVersionOverride: "1.20"
test-enterprise-legacy-label-values (stdout): Rendering helm test test-enterprise-legacy-label-values in PID 170: 'helm template test-enterprise-legacy-label-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-enterprise-legacy-label-values.yaml --output-dir operations/helm/manifests-intermediate/test-enterprise-legacy-label-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
scheduler-name-values (stdout):
scheduler-name-values (stdout): Templating scheduler-name-values
scheduler-name-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
scheduler-name-values (stdout): kubeVersionOverride: "1.20"
scheduler-name-values (stdout): Rendering helm test scheduler-name-values in PID 192: 'helm template scheduler-name-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/scheduler-name-values.yaml --output-dir operations/helm/manifests-intermediate/scheduler-name-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
gateway-enterprise-values (stdout):
gateway-enterprise-values (stdout): Templating gateway-enterprise-values
gateway-enterprise-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
gateway-enterprise-values (stdout): kubeVersionOverride: "1.20"
gateway-enterprise-values (stdout): Rendering helm test gateway-enterprise-values in PID 253: 'helm template gateway-enterprise-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/gateway-enterprise-values.yaml --output-dir operations/helm/manifests-intermediate/gateway-enterprise-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
test-enterprise-values (stdout):
test-enterprise-values (stdout): Templating test-enterprise-values
test-enterprise-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-enterprise-values (stdout): kubeVersionOverride: "1.20"
test-enterprise-values (stdout): Rendering helm test test-enterprise-values in PID 255: 'helm template test-enterprise-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-enterprise-values.yaml --output-dir operations/helm/manifests-intermediate/test-enterprise-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
test-enterprise-k8s-1.25-values (stdout):
test-enterprise-k8s-1.25-values (stdout): Templating test-enterprise-k8s-1.25-values
test-enterprise-k8s-1.25-values (stdout): Checking for kubeVersionOverride inside tests' values.yaml ...
test-enterprise-k8s-1.25-values (stdout): kubeVersionOverride: "1.25"
test-enterprise-k8s-1.25-values (stdout): Rendering helm test test-enterprise-k8s-1.25-values in PID 262: 'helm template test-enterprise-k8s-1.25-values operations/helm/charts/mimir-distributed -f operations/helm/charts/mimir-distributed/ci/offline/test-enterprise-k8s-1.25-values.yaml --output-dir operations/helm/manifests-intermediate/test-enterprise-k8s-1.25-values-generated --namespace citestns --set-string regoTestGenerateValues=true'
graphite-enabled-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
test-oss-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
graphite-enabled-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
test-oss-logical-multizone-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
metamonitoring-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
test-oss-logical-multizone-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
metamonitoring-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
test-enterprise-configmap-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
openshift-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
test-enterprise-configmap-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
openshift-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
test-oss-multizone-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
test-oss-multizone-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
test-oss-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
small-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
large-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
small-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
large-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
test-oss-topology-spread-constraints-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
gateway-nginx-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
test-oss-topology-spread-constraints-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
gateway-nginx-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
test-oss-k8s-1.25-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
test-oss-k8s-1.25-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
test-enterprise-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
test-oss-logical-multizone-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
test-enterprise-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
test-enterprise-k8s-1.25-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
test-enterprise-k8s-1.25-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
metamonitoring-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
test-oss-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
test-vault-agent-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
test-enterprise-configmap-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
test-vault-agent-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
graphite-enabled-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
openshift-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
test-oss-multizone-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
scheduler-name-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
scheduler-name-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
gateway-enterprise-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
large-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
test-enterprise-legacy-label-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/alerts.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/alerts.yaml. Contents of linked file included and used
gateway-enterprise-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
test-enterprise-legacy-label-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/dashboards resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/dashboards. Contents of linked file included and used
test-oss-topology-spread-constraints-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
test-oss-k8s-1.25-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
small-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
test-enterprise-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
gateway-nginx-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
test-enterprise-k8s-1.25-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
test-vault-agent-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
scheduler-name-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
gateway-enterprise-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
test-enterprise-legacy-label-values (stderr): walk.go:74: found symbolic link in path: /go/src/github.com/grafana/mimir/operations/helm/charts/mimir-distributed/mixins/rules.yaml resolves to /go/src/github.com/grafana/mimir/operations/mimir-mixin-compiled/rules.yaml. Contents of linked file included and used
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.admin-cache.topologySpreadConstraints (map[])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.admin_api.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.query_frontend.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.chunks-cache.topologySpreadConstraints (map[])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.query_scheduler.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.store_gateway.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.metadata-cache.topologySpreadConstraints (map[])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.index-cache.topologySpreadConstraints (map[])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.ingester.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.overrides_exporter.topologySpreadConstraints (map[])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.compactor.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.results-cache.topologySpreadConstraints (map[])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.querier.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.gateway.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.alertmanager.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.gateway.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.alertmanager.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.store_gateway.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.index-cache.topologySpreadConstraints (map[])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.querier.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.metadata-cache.topologySpreadConstraints (map[])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.admin_api.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.admin-cache.topologySpreadConstraints (map[])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.ingester.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.compactor.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.query_frontend.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.chunks-cache.topologySpreadConstraints (map[])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.query_scheduler.topologySpreadConstraints (map[maxSkew:1 topologyKey:kubernetes.io/hostname whenUnsatisfiable:ScheduleAnyway])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.overrides_exporter.topologySpreadConstraints (map[])
test-oss-topology-spread-constraints-values (stderr): coalesce.go:220: warning: cannot overwrite table with non table for mimir-distributed.results-cache.topologySpreadConstraints (map[])
gateway-nginx-values (stderr): Error: template: mimir-distributed/templates/store-gateway/store-gateway-pdb.yaml:1:4: executing "mimir-distributed/templates/store-gateway/store-gateway-pdb.yaml" at <include "mimir.lib.podDisruptionBudget" (dict "ctx" $ "component" "store-gateway" "memberlist" true)>: error calling include: template: mimir-distributed/templates/lib/pdb.tpl:10:15: executing "mimir.lib.podDisruptionBudget" at <include "mimir.podDisruptionBudget.apiVersion" $.ctx>: error calling include: template: mimir-distributed/templates/_helpers.tpl:381:9: executing "mimir.podDisruptionBudget.apiVersion" at <semverCompare ">= 1.21-0" (include "mimir.kubeVersion" .)>: error calling semverCompare: Invalid Semantic Version
gateway-nginx-values (stderr):
gateway-nginx-values (stderr): Use --debug flag to render out invalid YAML
helm template PID 198 exited with non-zero exit code 1. Aborting.
make: *** [Makefile:449: build-helm-tests] Error 1
make: Leaving directory '/go/src/github.com/grafana/mimir'

real	0m7.719s
user	0m0.019s
sys	0m0.038s
make: *** [build-helm-tests] Error 2
```

</p>
</details>